package/MDAnalysis/core/topologyattrs.py

```diff
@@ -2041,6 +2051,387 @@ def total_charge(group, compound='group'):
+                    * chgs[atom_mask][:,:,None], axis=1)
+
+        return dipole_vector
+
+    transplants[GroupBase].append(
+        ('dipole_vector', dipole_vector))
+
+    def dipole_moment(group, **kwargs):
+        r"""Dipole moment of the group or compounds in a group.
+
+        Computes the dipole moment of :class:`Atoms<Atom>` in the group.
+        Dipole per :class:`Residue`, :class:`Segment`, molecule, or
+        fragment can be obtained by setting the `compound` parameter
+        accordingly.
+
+        Note that when there is a net charge, the magnitude of the dipole 
+        moment is dependant on the ``center`` chosen. 
+        See :meth:`~dipole_vector`.
+
+        Parameters
+        ----------
+        group : obj
+            AtomGroup
+        wrap : bool, optional
+            If ``True`` and `compound` is ``'group'``, move all atoms to the
+            primary unit cell before calculation.
+            If ``True`` and `compound` is not ``group``, the
+            centers of mass of each compound will be calculated without moving
+            any atoms to keep the compounds intact.
+        unwrap : bool, optional
+            If ``True``, compounds will be unwrapped before computing their
+            centers.
+        compound : {'group', 'segments', 'residues', 'molecules', \
+                    'fragments'}, optional
+            If ``'group'``, a single dipole vector returns. Otherwise, an
+            array of each :class:`Segment`, :class:`Residue`, molecule, or
+            fragment will be returned as an array of position vectors, i.e.
+            a 2d array.
+            Note that, in any case, *only* the positions of
+            :class:`Atoms<Atom>` *belonging to the group* will be taken into
+            account.
+        center : str, optional
+            Choose whether the dipole moment is calculated at the center of 
+            "mass" or the center of "charge"
+
+        Returns
+        -------
+        numpy.ndarray
+            Dipole moment(s) of (compounds of) the group in :math:`eÅ`.
+            If `compound` was set to ``'group'``, the output will be a single
+            value. Otherwise, the output will be a 1d array of shape ``(n,)``
+            where ``n`` is the number of compounds.
+
+
+        .. versionadded:: 2.4.0
+        """
+
+        atomgroup = group.atoms
+
+        dipole_vector = atomgroup.dipole_vector(**kwargs)
+
+        if len(dipole_vector.shape) > 1:
+            dipole_moment = np.sqrt(np.sum(dipole_vector**2, axis=1))
+        else:
+            dipole_moment = np.sqrt(np.sum(dipole_vector**2))
+
+        return dipole_moment
+
+    transplants[GroupBase].append(
+        ('dipole_moment', dipole_moment))
+
+    @warn_if_not_unique
+    @_pbc_to_wrap
+    @check_wrap_and_unwrap
+    def quadrupole_tensor(group, wrap=False, unwrap=False, 
+        compound='group', center="mass"):
+        r"""Traceless quadrupole tensor of the group or compounds.
+         
```

Comment:
Please add the relevant equation.

Reply:
How would to like tensors and vectors to be denoted? Is a double bar and a bar respectively ok?

Reply:
Throughout we've been using bold face roman (\mathbf)  $\mathbf{v}$ for vectors. Simple 2D tensors/matrices are sans-serif (\mathsf) $\mathsf{A}$ so matrix-vector multiplication is 

$$
\mathbf{b} = \mathsf{A} \cdot \mathbf{v}.
$$

Would that work or be confusing for the equations that you want to write? (I would also recommend writing equations in components, such  as 

$$
b_{i} = A_{ij} v_{j}
$$ 

(Einstein summation) or 

$$
b_{i} = \sum_{j=1}^{N} A_{ij} v_{j}
$$

Reply:
Thanks!

Reply:
I added the equations, but it appears that greek letters (\mu) don't look different whether mathbf{} is added or not. What do you think?

Reply:
Use \boldmath https://texfaq.org/FAQ-boldgreek

Reply:
`$\boldsymbol{\mu}$` is the way to go